src/providers/virtualraster/qgsvirtualrasterprovider.cpp

```diff
@@ -336,6 +346,22 @@ int QgsVirtualRasterProvider::capabilities() const
   return capability;
 }
 
+QgsRasterIdentifyResult QgsVirtualRasterProvider::identify( const QgsPointXY &point, QgsRaster::IdentifyFormat format, const QgsRectangle &boundingBox, int width, int height, int dpi )
+{
+  auto result = QgsRasterDataProvider::identify( point, format, boundingBox, width, height, dpi );
+
+  // Checking the value returned by QgsRasterDataProvider::identify
+  // If it is NoData value, replace it with QVariant()
+  auto bandValues = result.results();
```

Comment:
I am a big fan of `auto` but other developers aren't, can you please replace `auto` with their actual types?

Reply:
done :)